README.md

```diff
@@ -132,32 +376,127 @@ comp.run_compression(
+
 ```
-ImportError: /[LONG_PATH]/synapse_compress_state.abi3.so: cannot allocate memory in static TLS block
+$ cd compressor_integration_tests/
+$ docker-compose up -d
+$ cargo test --workspace
+$ docker-compose down
 ```
-If this happens then try running the following:
+
+# Using the synapse_compress_state library
+
+If you want to use the compressor in another project, it is recomended that you
+use jemalloc `https://github.com/gnzlbg/jemallocator`. 
+
+To prevent the progress bars from being shown, use the `no-progress-bars` feature.
+(See `auto_compressor/Cargo.toml` for an example)
+
+# Troubleshooting
+
+## Connecting to database
+
+### From local machine
+
+If you setup Synapse using the instructions on https://matrix-org.github.io/synapse/latest/postgres.html
+you should have a username and password to use to login to the postgres database. To run the compressor
+from the machine where Postgres is running, the url will be the following:
+
+`postgresql://synapse_user:synapse_password@localhost/synapse`
+
+### From remote machine
+
+If you wish to connect from a different machine, you'll need to edit your Postgres settings to allow
+remote connections.
+
+To `/etc/postgresql/12/main/pg_hba.conf` add the following:
+
 ```
-LD_PATH=/[LONG_PATH]/synapse_compress_state.abi3.so ./my_python_script
+#   TYPE    DATABASE  USER            ADDRESS    METHOD
+    host    synapse   synapse_user    ADDR        md5
 ```
-Or just try disabling jemalloc:
+Substitute `ADDR` with the IP address of the machine you wish to connect from followed by `/32` (if it is
+an ipv4 address) or `/128` if it's an ipv6 address (e.g. `234.123.42.555/32` or `3a:23:5d::23:37/128`).
+
+If you want to be able to connect from any address (**for testing ONLY**) then you can use
+`0.0.0.0/0` or `::/0`
+
+To `/etc/postgresql/12/main/postgresql.conf` add the following:
+
 ```
-$ maturin develop --cargo-extra-args="--no-default-features"
+listen_addresses = 'localhost, IP_ADDR'
 ```
+Substitute `IP_ADDR` with your ip address (WITHOUT the `/32` or `/128` that was used in `pg_hba.conf`).
 
-## Running tests
-There are integration tests for these tool stored in `compressor_integration_tests/`
+If you want to allow connections from any address (**for testing ONLY**) then substitute `IP_ADDR` with `*`
 
-To run the integration tests, you first need to start up a postgres database
-for the libary to talk to. There is a docker-compose file that sets one up
-with all of the correct tables. The tests can therefore be run as follows:
+### Non default port
+
+By default, it tries to connect to a Postgres server running on port 5432. If you have configured your
+database to use a different port then the URL will take the following form:
+
+`postgresql://synapse_user:synapse_password@mydomain:PORT/synapse`
+
+See [the postgres crate documentation](https://docs.rs/tokio-postgres/0.7.2/tokio_postgres/config/struct.Config.html)
+for the full list of options.
```

Comment:
I don't think this section is helpful, or at least should be part of the generic "how do I connect to postgres"